### PR TITLE
Run build on deployment host

### LINKED ISSUE #10: Delegate Building the React Bundle to a Build Server
By using ansible [delegation](https://docs.ansible.com/ansible/latest/user_guide/playbooks_delegation.html#delegation), we can leave the task of [compiling the react bundle](https://github.com/onaio/ansible-react/blob/c8c27a53d07f2cacd2a24c2c8dada6dae8dc9462/tasks/configure.yml#L17) to another host(preferably the local ansible controller)

To do this we'll have to
- Install node on the host to build the bundle
- Install the node modules in package.json on the build host
- Copy environment variables to the build host
- Build the bundle on the build host
- Copy the output to the remote app server

### PR DESCRIPTION
**Pull request by morrismukiri**
- Use molecule to test the role
- Switch to running the package build on the ansible deployment host instead of the target host. This closes #10

### CODE REVIEW

File: tasks/install.yml [hunk 1]
```diff
@@ -1,48 +1,35 @@
...
-  shell: curl -sL https://deb.nodesource.com/setup_{{ react_node_version }} | sudo -E bash -
-  become: True
-  become_user: root
-  args:
-    warn: no
-
-- name: Update apt cache
-  apt:
-    update_cache: "yes"
+- name: Install NVM
```
Comment (morrismukiri) [inline tasks/install.yml]: We might skip installing NVM and make nvm on the deployment a prerequisite for the role to run.

**Comment by morrismukiri** [comment]
If yarn is not a hard requirement, we can use npm to run the build command, saving us the need to install an extra  package on the deployment host.

**Comment by moshthepitt** [comment]
> If yarn is not a hard requirement, we can use npm to run the build command, saving us the need to install an extra package on the deployment host.

@morrismukiri we should tread lightly here.  In some scenarios, especially with monorepos like js-tools and OpenSRP web, we cant do without yarn.

**Comment by morrismukiri** [comment]
> > If yarn is not a hard requirement, we can use npm to run the build command, saving us the need to install an extra package on the deployment host.
> 
> @morrismukiri we should tread lightly here. In some scenarios, especially with monorepos like js-tools and OpenSRP web, we cant do without yarn.

Ok. Then we can potentially install yarn using npm as a global package, so that we don't have to require super-user privileges to install yarn.

**Comment by moshthepitt** [comment]
> Ok. Then we can potentially install yarn using npm as a global package, so that we don't have to require super-user privileges to install yarn.

I'm fine with that.

**Comment by p-netm** [comment]
closed by #14

**Pull request closed without merging**